cfme/tests/infrastructure/test_vmware_provider.py

```diff
@@ -407,13 +407,96 @@ def test_esxi_reboot_not_orphan_vms(appliance, provider):
+        testSteps:
+            1.Add VMware provider to CFME
+            2.SSH to CFME appliance and perform following steps in rails console
+                '''
+                $evm = MiqAeMethodService::MiqAeService.new(MiqAeEngine::MiqAeWorkspaceRuntime.new)
+                p = $evm.vmdb(:ems_infra).first
+                p.class.name
+                p.switches.first.class.name
+                '''
+        expectedResults:
+            1.Provider added successfully and is refreshed
+            2.p.switches.first.class.name returns exit status 0(success) and lists class name
+            containing HostVirtualSwitch
+    """
+    command = """'$evm = MiqAeMethodService::MiqAeService.new(MiqAeEngine::MiqAeWorkspaceRuntime.new);\
+                p = $evm.vmdb(:ems_infra).first;\
+                p.class.name;\
+                puts "class name [#{p.switches.first.class.name}]"'
+                """
+    result = appliance.ssh_client.run_rails_command(command)
+    assert result.success, "SSH Command result was unsuccessful: {}".format(result)
+    logger.info("Output of Rails command was {}".format(result.output))
+    assert 'switch' in result.output.lower()
+
+
+@pytest.mark.tier(1)
+@pytest.mark.provider([VMwareProvider], selector=ONE_PER_TYPE, override=True)
+@pytest.mark.meta(automates=[1719399])
+def test_rebuilt_vcenter_duplicate_hosts(appliance, provider):
+    """
+    If vCenter rebuilt without removing from CFME, hosts should not be archived and
+    duplicate records should not be created.
+
+    Bugzilla:
+        1719399
+
+    Polarion:
+        assignee: kkulkarn
+        casecomponent: Infra
+        caseimportance: critical
+        initialEstimate: 1/2h
+        testtype: integration
+        testSteps:
+            1.Add VMware provider to CFME
+            2.Check hosts count on hosts page
+            3.SSH to CFME appliance & run given steps in rails console(to mimic vCenter rebuild)
+                '''
+                Host.all.each { |h| h.ems_id = nil; h.ems_ref = h.id.to_s; h.save! }
+                '''
+            4.Refresh provider and check hosts count/name.
+        expectedResults:
+            1.Provider added successfully and is refreshed
+            2.Note hosts count
+            3.Command runs successfully
+            4.Host count does not change, no duplicate hosts found.
+    """
+    hosts_before = len(provider.hosts.all())
+    command = "'Host.all.each { |h| h.ems_id = nil; h.ems_ref = h.id.to_s; h.save! }'"
+    result = appliance.ssh_client.run_rails_command(command)
+    assert result.success, "SSH Command result was unsuccessful: {}".format(result)
+    logger.info("Output of Rails command was {}".format(result.output))
+    provider.refresh_provider_relationships()
+    # Using appliance.rest_api as hosts.all() do not return archived hosts, I need those too
+    hosts_after = len(appliance.rest_api.collections.hosts.all)
+    assert hosts_before == hosts_after
```

Comment:
I think we are assuming that there are no archived hosts at the start of this test. If there is an archived host, this will fail.

Reply:
Discussed offline, updated the `hosts_before` call